src/core/net/srp_client.hpp

```diff
@@ -410,6 +410,27 @@ class Client : public InstanceLocator, private NonCopyable
      */
     void SetCallback(Callback aCallback, void *aContext);
 
+    /**
+     * This method gets the TTL used in SRP update requests.
+     *
+     * Note that this is the TTL that would be requested by the SRP client. Server may choose to accept a different TTL.
+     *
+     * @returns The TTL (in seconds).
+     *
+     */
+    uint32_t GetTtl(void) const { return mTtl ? mTtl : mLeaseInterval; }
```

Comment:
What if `mLeaseInterval` is shorter than specified `mTtl` ? 
Does it make sense to do something like `mTtl ? OT_MIN(mTtl, mLeaseInterval) : mLeaseInterval`?

Reply:
I checked the SRP draft. I think we want to use the min.

>    TTLs sent in SRP Updates are advisory: they indicate the SRP client's
>    guess as to what a good TTL would be.  SRP servers may override these
>    TTLs.  SRP servers SHOULD ensure that TTLs are reasonable: neither
>    too long nor too short.  **The TTL should never be longer than the
>    lease time** .

Reply:
Things get a bit more complicated when the user changes the lease interval too. I'm inclined to let the server handle it. - see https://datatracker.ietf.org/doc/html/draft-ietf-dnssd-srp#section-3. Thoughts?

Reply:
```
TTLs sent in SRP Updates are advisory: they indicate the SRP client's
guess as to what a good TTL would be. SRP servers may override these
TTLs. SRP servers SHOULD ensure that TTLs are reasonable: neither
too long nor too short. The TTL should never be longer than the
lease time .
```

I read this as the SRP server should be responsible for overriding the requested TTL value. So should we keep the client simple and let the server handle it?

Reply:
I agree. The server has the authority to actually accept the lease interval. The lease from clinet is again its desire.

Reply:
I am inclined to think that we should not send TTL larger than Lease Time in the first place, or even force it into the SRP client implementation. For example, we should not allow setting TTL that is larger than Lease Time. 

Thoughts?

Reply:
Both TTL and lease time in an SRP Update Request are proposals. Ultimately, the server gets to choose the values. If the server chooses a smaller lease time, then the server also needs to choose a smaller TTL. As a result, the SRP Server already needs to incorporate the logic of setting the TTL appropriately. It's not clear to me that there is much value in enforcing it on the client side.